runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/config/MqttKafkaBindingConfig.java

```diff
@@ -64,4 +102,58 @@ public String16FW retainedTopic()
     {
         return options.topics.retained;
     }
+
+    public List<MqttKafkaRouteConfig> bootstrapRoutes()
+    {
+        final List<MqttKafkaRouteConfig> bootstrapRoutes = new ArrayList<>();
+        routes.forEach(r ->
+        {
+            if (options.clients.stream().anyMatch(r::matchesClient))
+            {
+                bootstrapRoutes.add(r);
+            }
+        });
+        return bootstrapRoutes;
```

Comment:
This method is allocating on each call, suggest pre-allocating once in constructor instead if viable.